src/main/scala/ai/eto/rikai/sql/spark/RikaiSparkSessionExtensions.scala

```diff
@@ -14,12 +16,90 @@
 
 package ai.eto.rikai.sql.spark
 
+import ai.eto.rikai.sql.model.{Catalog, ModelSpec, Registry}
 import ai.eto.rikai.sql.spark.expressions.Predict
 import ai.eto.rikai.sql.spark.parser.{RikaiExtSqlParser, RikaiSparkSQLParser}
-import org.apache.spark.sql.SparkSessionExtensions
+import com.thoughtworks.enableIf
 import org.apache.spark.sql.catalyst.FunctionIdentifier
-import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionInfo}
+import org.apache.spark.sql.catalyst.analysis.{
+  UnresolvedAttribute,
+  UnresolvedFunction
+}
+import org.apache.spark.sql.catalyst.expressions.{
+  Expression,
+  ExpressionInfo,
+  Literal
+}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.rules.Rule
 import org.apache.spark.sql.rikai.expressions.{Area, IOU}
+import org.apache.spark.sql.{SparkSession, SparkSessionExtensions}
+
+private class MlPredictRule(val session: SparkSession)
+    extends Rule[LogicalPlan] {
+
+  val catalog = Catalog.getOrCreate(session)
```

Comment:
nit: call this modelCatalog so it's explicit (i.e., not the hive catalog) without having to look through the imports?

Reply:
will do